crux-test/test/crux/calcite_test.clj

```diff
@@ -0,0 +1,192 @@
+
+  (t/testing "order by"
+    (t/is (= [{:name "Ivan"}
+              {:name "Malcolm"}]
+             (query "SELECT PERSON.NAME FROM PERSON ORDER BY NAME ASC")))
+    (t/is (= [{:name "Malcolm"}
+              {:name "Ivan"}]
+             (query "SELECT PERSON.NAME FROM PERSON ORDER BY NAME DESC"))))
+
+  (t/testing "retrieve data"
+    (t/is (= [{:name "Ivan"}
+              {:name "Malcolm"}]
+             (query "SELECT PERSON.NAME FROM PERSON")))
+
+    (t/testing "retrieve data case insensitivity of table schema"
+      (t/is (= [{:name "Ivan"}
+                {:name "Malcolm"}]
+               (query "select person.name from person")))))
+
+  (t/testing "multiple columns"
+    (t/is (= [{:name "Ivan" :homeworld "Earth"}
+              {:name "Malcolm" :homeworld "Mars"}]
+             (query "SELECT PERSON.NAME,PERSON.HOMEWORLD FROM PERSON"))))
+
+  (t/testing "wildcard columns"
+    (t/is (= #{{:name "Ivan" :homeworld "Earth" :id ":ivan" :alive true}
+               {:name "Malcolm" :homeworld "Mars" :id ":malcolm" :alive false}}
+             (set (query "SELECT * FROM PERSON")))))
+
+  (t/testing "equals operand"
+    (t/is (= #{{:name "Ivan"}}
+             (set (query "SELECT NAME FROM PERSON WHERE NAME = 'Ivan'"))))
+    (t/is (= #{{:name "Malcolm"}}
+             (set (query "SELECT NAME FROM PERSON WHERE NAME <> 'Ivan'"))))
+    (t/is (= #{{:name "Ivan"}}
+             (set (query "SELECT NAME FROM PERSON WHERE 'Ivan' = NAME")))))
+
+  (t/testing "in operand"
+    (t/is (= #{{:name "Ivan"}}
+             (set (query "SELECT NAME FROM PERSON WHERE NAME in ('Ivan')")))))
+
+  (t/testing "and"
+    (t/is (= #{{:name "Ivan"}}
+             (set (query "SELECT NAME FROM PERSON WHERE NAME = 'Ivan' AND HOMEWORLD = 'Earth'")))))
+
+  (t/testing "or"
+    (t/is (= #{{:name "Ivan"} {:name "Malcolm"}}
+             (set (query "SELECT NAME FROM PERSON WHERE NAME = 'Ivan' OR NAME = 'Malcolm'")))))
+
+  (t/testing "boolean"
+    (t/is (= #{{:name "Ivan"}}
+             (set (query "SELECT NAME FROM PERSON WHERE ALIVE = TRUE"))))
+    (t/is (= #{{:name "Malcolm"}}
+             (set (query "SELECT NAME FROM PERSON WHERE ALIVE = FALSE")))))
+
+  (t/testing "like"
+    (t/is (= #{{:name "Ivan"}}
+             (set (query "SELECT NAME FROM PERSON WHERE NAME LIKE 'Iva%'"))))
+    (t/is (= #{{:name "Ivan"} {:name "Malcolm"}}
+             (set (query "SELECT NAME FROM PERSON WHERE NAME LIKE 'Iva%' OR NAME LIKE 'Mal%'")))))
+
+  (t/testing "arbitrary sql function"
+    (t/is (= #{{:name "Iva"}}
+             (set (query "SELECT SUBSTRING(NAME,1,3) AS NAME FROM PERSON WHERE NAME = 'Ivan'")))))
+
+  (t/testing "namespaced keywords"
+    (f/transact! *api* (f/people [{:crux.db/id :human/ivan :name "Ivan" :homeworld "Earth" :alive true}]))
+    (t/is (= [{:id ":human/ivan", :name "Ivan"}] (query "SELECT ID,NAME FROM PERSON WHERE ID = CRUXID('human/ivan')"))))
+
+  (t/testing "numeric values"
+    (fapi/delete-all-entities)
```

Comment:
the fixtures deliberately start up a fresh Crux node for every test - could we split this test out into two if it doesn't depend on the state from the first part? otherwise, you have to check every `t/testing` block to figure out the state of the database